content/blog/how-to-create-a-ci-cd-pipeline-to-execute-automation-test-suite-with-hpe-greenlake-for-containers.md

```diff
@@ -0,0 +1,81 @@
+  HPE GreenLake for Containers
+date: 2022-06-08T06:09:47.137Z
+featuredBlog: false
+priority: 1
+author: Sweta Katkoria
+authorimage: /img/swetakatkoria_photo.jpg
+thumbnailimage: /img/thumbnail.jpeg
+tags:
+  - hpe-greenlake
+  - Automation
+  - Grafana
+---
+## Introduction
+
+*[HPE GreenLake for Containers](https://www.hpe.com/us/en/greenlake/containers.html)*, one of the HPE GreenLake Cloud Services, is built upon [HPE Ezmeral Runtime Enterprise](https://www.hpe.com/us/en/software/ezmeral-runtime.html) and runs via underlying container-based opensource infrastructure Kubernetes. The HPE GreenLake Central dashboard allows performing several cluster-related operations via navigating to the *Clusters* module. This module includes operations such as cluster creation by using default and custom machine blueprints, cluster deletion, cluster scaling up, and cluster scaling down. It can navigate to the HPE Ezmeral Runtime Enterprise dashboard where an end-user can deploy various applications and fulfill the goal of reaching the containerization.
+
+Before proceeding to the app deployment phase, an end-user may be interested to perform some primary data verifications that can be helpful in various ways, for example, sanity testing, regression testing, knowing infrastructure health, and finding on-prem issues at the early stage. As a part of the smart automated solution, the user needs to verify the data such as the cluster health, cluster status, node status, host IP, CPU, and memory allocation. This blog can guide to achieve one of the ways to implement the Automation Pipeline via using automation weapons like Katalon as an automation testing software tool, CircleCI as continuous integration and continuous delivery platform, and Grafana as an interactive visualization web application tool that uses time-series data to deploy meaningful graphs.
+
+
+
+## How to Use Automation Pipeline Tools?
+
+In Katalon, Test cases can be structured using test suites with environment variables. Test execution can be parameterized and parallelized using profiles. Remote execution in Katalon Studio can be triggered by CI systems via Docker container or command-line interface. Automation job can be triggered for Cluster Creation operation, following via Cluster Scale Up, Cluster Scale Down, and Cluster Deletion operation, 
+
+The scripts for the above operations include verification points and required performance metrics. The automation suite starts recording the required time for the cluster to become ready upon the trigger of the cluster creation process. Similar way other cluster operations related data can be collected. Katalon Studio provides HTML-based reports or console logs to view the data after execution has been done. Any test script can help to extract the required data in form of a plain text-based file like .csv. CircleCi provides functionality to export this .csv file as an artifact inside the job. 
+
+Sample skeleton of circle-ci configuration:
```

Comment:
In addition to the CicleCI configuration script below, would you have any scripts or screenshots about Katalon to illustrate your CI/CD pipeline.